src/tls.c

```diff
@@ -8349,13 +8630,30 @@ int TLSX_KeyShare_Establish(WOLFSSL *ssl, int* doHelloRetry)
     }
 
     list = NULL;
-    /* Generate a new key pair. */
+    /* Generate a new key pair except in the case of OQS KEM because we
+     * are going to encapsulate and that does not require us to generate a
+     * key pair.
+     */
     ret = TLSX_KeyShare_New(&list, clientKSE->group, ssl->heap, &serverKSE);
     if (ret != 0)
         return ret;
 
     if (clientKSE->key == NULL) {
-        ret = TLSX_KeyShare_GenKey(ssl, serverKSE);
+#ifdef HAVE_LIBOQS
+        if (clientKSE->group < WOLFSSL_OQS_MIN ||
+            clientKSE->group > WOLFSSL_OQS_MAX ) {
+#endif
+            ret = TLSX_KeyShare_GenKey(ssl, serverKSE);
+#ifdef HAVE_LIBOQS
```

Comment:
I prefer the braces not to be cross with macros. Can you have the end brace then `#ifdef HAVE_LIBOQS` and `else {` ?

Reply:
I think I've come up with something a bit neater.